File: DDDCore.NET/Entity.cs
using System;
using System.Collections.Generic;

namespace DDD.Domain.Core;

public abstract class Entity : ValueObject
{
    public Guid Id { get; private set; }

    public Entity(Guid id)
    {
        Id = id;
    }

    protected override IEnumerable<object> GetValues()
    {
        yield return Id;
    }
}

File: DDDCore.Java/Entity.java
package com.infosupport.ddd.domain.core;

import java.util.UUID;

public abstract class Entity extends ValueObject {

    private final UUID _id;
    
    public Entity(UUID id) {
        _id = id;
    }
    
    public UUID getId() {
        return _id;
    }
}
